csharp/src/Apache.Arrow/RecordBatch.cs

```diff
@@ -40,6 +42,24 @@ public IArrowArray Column(string columnName)
             return _arrays[fieldIndex];
         }
 
+        public void Dispose()
```

Comment:
are there any types that derive from `RecordBatch`? Do we need full dispose pattern implementation here?

Reply:
It isn't sealed, and it has a `public` constructor, so I felt it was best. If we seal the class, we can remove the full dispose pattern.